project/Unidoc.scala

```diff
@@ -7,47 +7,44 @@ import sbt.Keys._
 import sbt.File
 import scala.annotation.tailrec
 
-object Unidoc {
+object Scaladoc extends AutoPlugin {
 
-  def settings(ignoreAggregates: Seq[Project], ignoreProjects: Seq[Project]) = {
-    val withoutAggregates = ignoreAggregates.foldLeft(inAnyProject) { _ -- inAggregates(_, transitive = true, includeRoot = true) }
-    val docProjectFilter = ignoreProjects.foldLeft(withoutAggregates) { _ -- inProjects(_) }
-
-    inTask(unidoc)(Seq(
-      unidocProjectFilter in ScalaUnidoc := docProjectFilter,
-      unidocProjectFilter in JavaUnidoc := docProjectFilter,
-      apiMappings in ScalaUnidoc := (apiMappings in (Compile, doc)).value
-    ))
+  object CliOption {
+    val genjavadocEnabled = sys.props.getOrElse("akka.genjavadoc.enabled", "false") == "true"
+    val scaladocDiagramsEnabled = sys.props.getOrElse("akka.scaladoc.diagrams", "true") == "true"
+    val scaladocAutoAPI = sys.props.getOrElse("akka.scaladoc.autoapi", "true") == "true"
   }
 
-  val genjavadocEnabled = sys.props.get("akka.genjavadoc.enabled").getOrElse("false").toBoolean
-  val (unidocSettings, javadocSettings) =
-    if (genjavadocEnabled)
-      (scalaJavaUnidocSettings, genjavadocExtraSettings ++ Seq(
-        scalacOptions in Compile += "-P:genjavadoc:fabricateParams=true",
-        unidocGenjavadocVersion in Global := "0.9"))
-    else (scalaUnidocSettings, Nil)
+  override def trigger = allRequirements
+  override def requires = plugins.JvmPlugin
 
-  lazy val scaladocDiagramsEnabled = sys.props.get("akka.scaladoc.diagrams").getOrElse("true").toBoolean
-  lazy val scaladocAutoAPI = sys.props.get("akka.scaladoc.autoapi").getOrElse("true").toBoolean
+  override lazy val projectSettings = scaladocSettings ++ javadocSettings
 
-  def scaladocSettings: Seq[sbt.Setting[_]] = {
-    scaladocSettingsNoVerificationOfDiagrams ++
-      (if (scaladocDiagramsEnabled) Seq(doc in Compile ~= scaladocVerifier) else Seq.empty)
-  }
+  val validateDiagrams = settingKey[Boolean]("Validate generated scaladoc diagrams")
+
+  val javadocSettings = if (CliOption.genjavadocEnabled) Seq(
+    scalacOptions in Compile += "-P:genjavadoc:fabricateParams=true",
```

Comment:
Sorry, the comment was supposed to be "need for `genjavadocExtraSettings`" it was there but now it's gone